bigtable/tests/instance_admin_integration_test.cc

```diff
@@ -193,37 +189,91 @@ TEST_F(InstanceAdminIntegrationTest, DeleteInstancesTest) {
 
 /// @test Verify that InstanceAdmin::ListClusters works as expected.
 TEST_F(InstanceAdminIntegrationTest, ListClustersTest) {
-  // The emulator does not support cluster operations.
-  if (UsingCloudBigtableEmulator()) {
-    return;
-  }
-  std::string id = "list-clusters-test";
+  std::string id =
+      "it-" + bigtable::testing::Sample(generator_, 8,
+                                        "abcdefghijklmnopqrstuvwxyz0123456789");
+
   bigtable::InstanceId instance_id(id);
+  bigtable::ClusterId cluster_id(id);
   bigtable::DisplayName display_name(id);
-  std::vector<std::pair<std::string, bigtable::ClusterConfig>> clusters_config;
-  clusters_config.push_back(std::make_pair(
-      id + "-cluster1", bigtable::ClusterConfig("us-central1-f", 0,
-                                                bigtable::ClusterConfig::HDD)));
+
+  bigtable::ClusterConfig cluster_config =
+      bigtable::ClusterConfig("us-central1-f", 3, bigtable::ClusterConfig::HDD);
+  auto cluster =
+      instance_admin_->CreateCluster(cluster_config, instance_id, cluster_id);
+
+  std::vector<std::pair<std::string, bigtable::ClusterConfig>>
+      clusters_config_list;
+  clusters_config_list.push_back(
+      std::make_pair(cluster_id.get(), cluster_config));
   auto instance_config =
-      bigtable::InstanceConfig(instance_id, display_name, clusters_config)
+      bigtable::InstanceConfig(instance_id, display_name, clusters_config_list)
           .set_type(bigtable::InstanceConfig::DEVELOPMENT);
-  auto instance_details =
-      instance_admin_->CreateInstance(instance_config).get();
-
-  // Create clusters in an instance
-  // TODO(#422) - Implement InstanceAdmin::CreateCluster
+  auto instance = instance_admin_->CreateInstance(instance_config).get();
 
-  // TODO(#418) - create an instance and test that its cluster is returned here.
-  auto clusters = instance_admin_->ListClusters(id);
-  for (auto const& i : clusters) {
+  auto clusters = instance_admin_->ListClusters(instance_id.get());
+  for (auto const& cluster : clusters) {
     auto const npos = std::string::npos;
-    EXPECT_NE(npos, i.name().find(instance_admin_->project_name()));
+    EXPECT_NE(npos, cluster.name().find(instance_admin_->project_name()));
+    EXPECT_NE(npos, cluster.name().find(instance_id.get()));
   }
   EXPECT_FALSE(clusters.empty());
```

Comment:
This fails in the integration tests against the emulator because CreateInstance (in the emulator) does not populate clusters_.  You need to fix the emulator, see #610 and talk to @manish-qlogic to avoid duplicating work.